File: pytools/examples/csharp/client/Protocol/MsgTestPhp.cs
using System.Collections;
using System.Collections.Generic;


public class MsgTestPhp
{
	private ushort _u16;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteUshort(this._u16);
		return packet;
	}

	public MsgTestPhp()
	{
	}

	public MsgTestPhp(Packet packet)
	{
		this._u16 = packet.ReadUshort();
	}

	public List<byte> GetBuffer()
	{
		return this.Encode().GetBuffer();
	}


	public ushort u16
	{
		get { return this._u16; }
		set { this._u16 = value; }
	}

}

File: pytools/examples/php/client/protocol/MsgTestPhp.php
<?php
namespace protocol;

use network\Packet;


class MsgTestPhp
{
	private $u16 = 0;


	public function __construct(&$packet=null)
	{
		if ($packet)
		{
			$this->decode($packet);
		}
	}

	public function decode(&$packet)
	{
		$this->u16 = $packet->readU16();
	}

	public function encode()
	{
		$packet = new Packet();
		$packet->writeU16($this->u16);
		return $packet->readBytes();
	}

	public function getBytes()
	{
		return $this->encode();
	}


	public function setU16($u16)
	{
		$this->u16 = $u16;
	}
	public function getU16()
	{
		return $this->u16;
	}

}
